src/platform/Darwin/DnssdHostNameRegistrar.cpp

```diff
@@ -17,186 +17,371 @@
+    {
+        if (interfaceId == inet6Interface.first)
+        {
+            char addr[INET6_ADDRSTRLEN] = {};
+            inet_ntop(AF_INET6, &inet6Interface.second, addr, sizeof(addr));
+            ChipLogDetail(Discovery, "\t\t* ipv6: %s", addr);
+        }
+    }
+}
 
-    DNSRecordRef dnsRecordRef;
-    return DNSServiceRegisterRecord(sdRef, &dnsRecordRef, kRegisterRecordFlags, mInterfaceId, hostname, rrtype, kDNSServiceClass_IN,
-                                    rdlen, rdata, 0, OnRegisterRecord, sdCtx);
+void LogDetails(nw_path_t path)
+{
+    auto status = nw_path_get_status(path);
+    ChipLogDetail(Discovery, "Status: %s", GetPathStatusString(status));
 }
 
-bool InterfaceRegistrant::HasValidFlags(unsigned int flags)
+void LogDetails(InetInterfacesVector inetInterfaces, Inet6InterfacesVector inet6Interfaces)
 {
-    return !(flags & IFF_POINTOPOINT) && (flags & IFF_RUNNING) && (flags & IFF_MULTICAST);
+    std::set<uint32_t> interfaceIds;
+    for (auto & inetInterface : inetInterfaces)
+    {
+        interfaceIds.insert(inetInterface.first);
+    }
+
+    for (auto & inet6Interface : inet6Interfaces)
+    {
+        interfaceIds.insert(inet6Interface.first);
+    }
+
+    for (auto interfaceId : interfaceIds)
+    {
+        char interfaceName[IFNAMSIZ] = {};
+        if_indextoname(interfaceId, interfaceName);
+        ChipLogDetail(Discovery, "\t%s (%u)", interfaceName, interfaceId);
+        LogDetails(interfaceId, inetInterfaces, inet6Interfaces);
+    }
 }
 
-bool InterfaceRegistrant::HasValidType(Inet::IPAddressType addressType, const sa_family_t addressFamily)
+void LogDetails(nw_interface_t interface, InetInterfacesVector inetInterfaces, Inet6InterfacesVector inet6Interfaces)
 {
-    bool useIPv4 = addressType == Inet::IPAddressType::kIPv4 || addressType == Inet::IPAddressType::kAny;
-    bool useIPv6 = addressType == Inet::IPAddressType::kIPv6 || addressType == Inet::IPAddressType::kAny;
+    auto interfaceId   = nw_interface_get_index(interface);
+    auto interfaceName = nw_interface_get_name(interface);
+    auto interfaceType = nw_interface_get_type(interface);
+    ChipLogDetail(Discovery, "\t%s (%u / %s)", interfaceName, interfaceId, GetInterfaceTypeString(interfaceType));
+    LogDetails(interfaceId, inetInterfaces, inet6Interfaces);
+}
+#endif // CHIP_DETAIL_LOGGING
 
-    return (useIPv6 && addressFamily == AF_INET6) || (useIPv4 && addressFamily == AF_INET);
+bool HasValidFlags(unsigned int flags, bool allowLoopbackOnly)
+{
+    VerifyOrReturnValue(!allowLoopbackOnly || (flags & IFF_LOOPBACK), false);
+    VerifyOrReturnValue(!(flags & IFF_POINTOPOINT), false);
+    VerifyOrReturnValue((flags & IFF_RUNNING), false);
+    VerifyOrReturnValue((flags & IFF_MULTICAST), false);
+    return true;
+}
+
+bool HasValidNetworkType(nw_interface_t interface)
+{
+    auto interfaceType = nw_interface_get_type(interface);
+    return interfaceType == nw_interface_type_wifi || interfaceType == nw_interface_type_wired;
```

Comment:
Will Thread return wifi here, do you know?

Reply:
I don't know. I can allow other interfaces if needed (such as `Other`). I'm mostly trying to avoid the `Cellular` type.

Reply:
I think we can change this in a follow-up, too.  Just need to make sure it gets tested over Thread...

Reply:
I am asking around for the information. Will fix that once I have the information.